src/nvim/regexp.c

```diff
@@ -2558,15 +2558,14 @@ long vim_regexec_multi(regmmatch_T *rmp, win_T *win, buf_T *buf, linenr_T lnum,
   }
   rex_in_use = true;
 
-  int result = rmp->regprog->engine->regexec_multi(rmp, win, buf, lnum, col,
-                                                   tm, timed_out);
+  int result = (int)rmp->regprog->engine->regexec_multi(rmp, win, buf, lnum, col, tm, timed_out);
```

Comment:
It may make more sense to make this `result` a `long`, as the declaration is already in the middle.